### PR TITLE
Closes #86: Setup GitHub Actions for Frontend & Backend

### LINKED ISSUE #86: Setup GitHub Actions
**Is your feature request related to a problem? Please describe.**
We would like setup CI/CD to automate some of the development/PR workflow. 

**Describe the solution you'd like**
Setup GitHub actions to automatically test and run linter/prettier

**Describe alternatives you've considered**
None

**Additional context**
Need to add the lint script to the frontend so it can automatically run the linter when we make a PR.

### PR DESCRIPTION
**Pull request by ambiyu**
Closes #86

Setup GitHub Actions for both frontend and backend. Automatically runs ESLint/Prettier and tests.

### CODE REVIEW

File: .github/workflows/backend-test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    name: Running Backend Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
```
Comment (RawiriHohepa) [inline .github/workflows/backend-test.yml]: Can you put in names for these steps similar to the lint files?

File: .github/workflows/backend-test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  test:
+    name: Running Backend Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (RawiriHohepa) [inline .github/workflows/backend-test.yml]: Why does this use matrix.node-version but the lint files define the version number?

**Review by RawiriHohepa — CHANGES_REQUESTED** [review]
Just a couple changes and wanting to know design decisions

File: .github/workflows/backend-test.yml (hunk 2 above)
Comment (ambiyu) [reply .github/workflows/backend-test.yml]: Forgot to change this after pasting from a sample 😦 
Changed to use Node.js v14

**Review by RawiriHohepa — APPROVED** [review]
Looks good now :)

**Review by RawiriHohepa — APPROVED** [review]
Good idea to combine the lint and testing actions, makes it much more efficient!

**Review by khal745 — APPROVED** [review]
LGTM! Thanks for setting this up!

**Review by a-158w — APPROVED** [review]
LGTM

**Pull request merged**